internal/database/testing/dqlitesuite.go

```diff
@@ -240,9 +242,9 @@ func (s *DqliteSuite) cleanupDB(c *gc.C, namespace string, db *sql.DB) {
 	defer s.mutex.Unlock()
 
 	if s.references == nil {
-		s.references = make(map[string]*sql.DB)
+		s.references = make(map[string][]*sql.DB)
 	}
-	s.references[namespace] = db
+	s.references[namespace] = append(s.references[namespace], db)
```

Comment:
This is a bit sad, the fact people are getting the same namespace over and over again, is a bit sad. I can't close the original one as it's still being used, so we have to sum them up.

Reply:
Fixed, we now pass back the instance for the same namespace.